test/tar_interpreter_test.go

```diff
@@ -0,0 +1,156 @@
+package test
+
+import (
+	"archive/tar"
+	"bytes"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/wal-g/wal-g/internal"
+)
+
+type testInterpretOptions struct {
+	DBDataDirectory string
+
+	Name     string
+	Typeflag byte
+
+	Create func(string) error
+	Delete func(string) error
+}
+
+func testInterpret(t *testing.T, options testInterpretOptions) (os.FileInfo, os.FileInfo) {
```

Comment:
You don't cover cases when an error occurs.

Reply:
You can use assert.IsType for it.

Reply:
I write this function only for **happy path** cases. I'll deal with **bad path** later.

Reply:
Ok, thanks. We will wait for your next PR :)